crates/phala-types/src/lib.rs

```diff
@@ -621,6 +623,27 @@ pub mod messaging {
 
 // Types used in storage
 
+#[derive(Encode, Decode, TypeInfo, Debug, Clone, PartialEq, Eq)]
+pub enum AttestationReport {
+    SgxIas {
+        ra_report: Vec<u8>,
+        signature: Vec<u8>,
+        raw_signing_cert: Vec<u8>,
+    },
+    OptOut,
+}
+
+#[derive(Encode, Decode, TypeInfo, Debug, Copy, Clone, PartialEq, Eq)]
+pub enum AttestationProvider {
+    Ias,
+    OptOut,
+    RootOrCouncil,
```

Comment:
Is it better to rename it to:
```
pub enum AttestationProvider {
  None,
  Root,
  Ias,
  Dcap,
}
```

or just
```
pub enum AttestationProvider {
  Root,
  Ias,
  Dcap,
}

let use_provider: Option<AttestationProvider>;
```

Reply:
will do

Reply:
Rnamed `OptOut` to `None` (the first solution)

Reply:
I meant several changes:
1. Rename `OptOut` -> `None`. (Done)
2. Let the `None` be the first or last variant, because it's special.
3. Is it OK to remove the Unknown and default to None?
4. Rename `RootOrCouncil` to be consistent with `ensure_root!`.

Reply:
Done